bigframes/core/__init__.py

```diff
@@ -49,470 +40,196 @@
-        ):
-            expr = self._hide_column(ordering_column)
-
-        expr_builder = expr.builder()
-        remain_cols = [
-            column for column in expr.columns if column.get_name() not in columns
-        ]
-        expr_builder.columns = remain_cols
-        return expr_builder.build()
-
-    def get_column_type(self, key: str) -> bigframes.dtypes.Dtype:
-        ibis_type = typing.cast(
-            bigframes.dtypes.IbisDtype, self._get_any_column(key).type()
+        table_expression = self.session.ibis_client.table(
+            f"{destination.project}.{destination.dataset_id}.{destination.table_id}"
         )
-        return typing.cast(
-            bigframes.dtypes.Dtype,
-            bigframes.dtypes.ibis_dtype_to_bigframes_dtype(ibis_type),
+        new_columns = [table_expression[column] for column in compiled.column_ids]
+        new_hidden_columns = [
+            table_expression[column]
+            for column in compiled._hidden_ordering_column_names
+        ]
+        return ArrayValue.from_ibis(
+            self.session,
+            table_expression,
+            columns=new_columns,
+            hidden_ordering_columns=new_hidden_columns,
+            ordering=compiled._ordering,
         )
 
-    def _get_ibis_column(self, key: str) -> ibis_types.Value:
-        """Gets the Ibis expression for a given column."""
-        if key not in self.column_ids:
-            raise ValueError(
-                "Column name {} not in set of values: {}".format(key, self.column_ids)
-            )
-        return typing.cast(ibis_types.Value, self._column_names[key])
+    def _get_table_size(self, destination_table):
+        return self.session._get_table_size(destination_table)
 
-    def _get_any_column(self, key: str) -> ibis_types.Value:
-        """Gets the Ibis expression for a given column. Will also get hidden columns."""
-        all_columns = {**self._column_names, **self._hidden_ordering_column_names}
-        if key not in all_columns.keys():
-            raise ValueError(
-                "Column name {} not in set of values: {}".format(
-                    key, all_columns.keys()
-                )
-            )
-        return typing.cast(ibis_types.Value, all_columns[key])
+    # Operations
 
-    def _get_hidden_ordering_column(self, key: str) -> ibis_types.Column:
-        """Gets the Ibis expression for a given hidden column."""
-        if key not in self._hidden_ordering_column_names.keys():
-            raise ValueError(
-                "Column name {} not in set of values: {}".format(
-                    key, self._hidden_ordering_column_names.keys()
-                )
-            )
-        return typing.cast(ibis_types.Column, self._hidden_ordering_column_names[key])
+    def drop_columns(self, columns: Iterable[str]) -> ArrayValue:
+        return ArrayValue(
+            nodes.DropColumnsNode(child=self.node, columns=tuple(columns))
```

Comment:
Add to backlog: just one Selection node type. (Though the ibis devs seemed to be going the other direction)

Might be good to follow substrait's architecture for this.

Reply:
created task